applications/PfemFluidDynamicsApplication/custom_processes/adaptive_time_interval_process.hpp

```diff
@@ -284,163 +284,148 @@ namespace Kratos
 	OpenMPUtils::PartitionedIterators(mrModelPart.Elements(),ElemBegin,ElemEnd);
 	for ( ModelPart::ElementIterator itElem = ElemBegin; itElem != ElemEnd; ++itElem )
 	  {
-	    unsigned int size = itElem->GetGeometry().size();
-	    unsigned int rigid_nodes   = 0;
-	    for(unsigned int i=0; i<size; i++)
-	      {
-		if(itElem->GetGeometry()[i].Is(RIGID)){
-		    rigid_nodes++;
+	    double temporaryTimeInterval=rCurrentProcessInfo[DELTA_TIME];
+	    double currentElementalArea =  0;
+	    const unsigned int dimension = (itElem)->GetGeometry().WorkingSpaceDimension();
+	    if(dimension==2){
+	      currentElementalArea =  (itElem)->GetGeometry().Area();
+	      Geometry<Node<3> >  updatedElementCoordinates;
+	      bool solidElement=false;
+	      for(unsigned int i=0; i<itElem->GetGeometry().size(); i++)
+		{
+		  if(itElem->GetGeometry()[i].Is(SOLID) || itElem->GetGeometry()[i].Is(TO_ERASE) || itElem->IsNot(ACTIVE)){
+		    solidElement=true;
+		  }
+	
+		  const array_1d<double,3> &Vel = itElem->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY);
+		  Point updatedNodalCoordinates=itElem->GetGeometry()[i].Coordinates()+Vel*temporaryTimeInterval;
```

Comment:
this operation assumes you are using a certain scheme, right?

Reply:
yes